cmd/soroban-rpc/lib/preflight/src/lib.rs

```diff
@@ -185,7 +188,7 @@ fn preflight_host_function_or_maybe_panic(
     let auth_payloads = host.get_recorded_auth_payloads()?;
 
     // Recover, convert and return the storage footprint and other values to C.
-    let (storage, budget, _) = host.try_finish().unwrap();
+    let (storage, budget, events) = host.try_finish().unwrap();
```

Comment:
Side question: Does this include diagnostic events? or are those emitted somewhere else?

Reply:
Good question. Now that you bring it up, I am not sure. @sisuresh @jayz22 ?

Reply:
Yeah these include diagnostic events.

Reply:
great!